Emit the nvidia driver version in gpu e2e test

chardch opened the pull request:
**What type of PR is this?**
> /kind cleanup

**What this PR does / why we need it**:
This PR logs the nvidia gpu driver version so that the test logs can be used to more easily verify the driver version is mapped to the correct kubernetes version. `nvidia-smi` is used in order to display the driver information, which contains the driver version. https://developer.nvidia.com/nvidia-system-management-interface


**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

chardch commented:
/cc @jiayingz

wgliang commented:
/ok-to-test

neolit123 commented on test/e2e/scheduling/nvidia-gpus.go:
```diff
@@ -180,6 +182,10 @@ func testNvidiaGPUs(f *framework.Framework) {
+	// Get  the Nvidia driver version
+	versionInfo, err := ioutil.ReadFile(driverVersionPath)
+	framework.Logf(string(versionInfo))
+	Expect(err).NotTo(HaveOccurred())
```
please use:
```
versionInfo, err := ioutil.ReadFile(driverVersionPath)
framework.ExpectNoError(err, "Failed to read driver version from %s", driverVersionPath)
framework.Logf(string(versionInfo))
```

oomichi replied:
+1, nice catch

chardch replied:
Thanks for catching this. Updated

neolit123 reviewed (commented):
/kind feature
/priority backlog

neolit123 commented:
the change as is seems to be failing:
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/77100/pull-kubernetes-e2e-gce-device-plugin-gpu/1121601780044533764/

oomichi commented:
/cc @oomichi

chardch commented:
/test pull-kubernetes-e2e-gce-device-plugin-gpu

chardch commented:
It looks like the file `/proc/driver/nvidia/version` doesn't exist on the test host machine. 

Is the nvidia driver information located elsewhere? Or is the code run on the cluster master, which doesn't have GPUs installed?

chardch committed:
Output the nvidia gpu information in the E2E test.

Including the gpu information simplifies driver version verification.
nvidia-smi is used in order to display gpu information, which contains the driver version.

chardch commented:
Changed this to output the driver information from within the container logs.

chardch commented:
/test pull-kubernetes-e2e-gce-device-plugin-gpu
/test pull-kubernetes-e2e-gce
/test pull-kubernetes-e2e-gce-100-performance

oomichi commented on test/e2e/scheduling/nvidia-gpus.go:
```diff
@@ -182,6 +183,14 @@ func testNvidiaGPUs(f *framework.Framework) {
+func logContainers(f *framework.Framework, pod *v1.Pod) {
+	for _, container := range pod.Spec.Containers {
+		logs, err := framework.GetPodLogs(f.ClientSet, f.Namespace.Name, pod.Name, container.Name)
+		framework.ExpectNoError(err, "Should be able to get container logs")
```
better to output `container.Name` also into the error message because this is used for multiple containers and it could be helpful if we face an issue.

chardch replied:
Thanks. That makes sense. Added the container name to the error message.

oomichi reviewed (commented):
Thanks for updating.

/lgtm

chardch commented:
/assign @listx 
Hi Linus, could you take a look at this PR when you have time?

chardch commented on test/images/cuda-vector-add/Dockerfile:
```diff
@@ -23,4 +23,4 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
 RUN make
 
-CMD ./vectorAdd
+CMD nvidia-smi && ./vectorAdd
```
Thanks for noting. I updated the commit message.

listx replied:
Looks like you updated the PR description, not the commit message itself? Can you `git amend` and repush?

chardch replied:
Ah got it thanks. Updated.

oomichi commented:
All comments seem to be solved.

/lgtm

listx commented:
/lgtm
/approve

chardch commented:
/assign @oomichi

oomichi commented:
/approve

oomichi commented:
/lgtm
